File: Tutorials/WinForms/DevExpress.MVVM/CS/Views/CustomersListView.cs
using System.Windows.Forms;
using DevExpress.Utils.MVVM.UI;
using DevExpress.XtraGrid.Views.Grid;
using WinFormsApplication.Services;
using WinFormsApplication.ViewModels;

namespace WinFormsApplication.Views {
    [ViewType("CustomerListView")]
    public partial class CustomersListView : DevExpress.XtraBars.Ribbon.RibbonForm {
        public CustomersListView() {
            InitializeComponent();
            if(!mvvmContext1.IsDesignMode)
                InitializeBindings();
        }
        void InitializeBindings() {
            mvvmContext1.RegisterService(new InstantFeedbackService(CustomersGridView));
            //
            var fluentApi = mvvmContext1.OfType<CustomerListViewModel>();
            fluentApi.WithEvent(this, nameof(Load))
                .EventToCommand(x => x.Reload());
            fluentApi.SetBinding(CustomersGridControl, bs => bs.DataSource, x => x.Customers);
            fluentApi.WithEvent<RowClickEventArgs>(CustomersGridView, nameof(CustomersGridView.RowClick))
                .EventToCommand(x => x.Edit(),
                    (RowClickEventArgs args) => args.Clicks == 2 && args.Button == MouseButtons.Left);
            fluentApi.BindCommand(btnNew, x => x.Add());
            fluentApi.BindCommand(btnDelete, x => x.Delete());
        }
    }
}

File: Tutorials/WinForms/DevExpress.MVVM/VB/Views/CustomersListView.vb
Imports System.Windows.Forms
Imports DevExpress.Utils.MVVM.UI
Imports DevExpress.XtraGrid.Views.Grid
Imports WinFormsApplication.Services
Imports WinFormsApplication.ViewModels

Namespace WinFormsApplication.Views
	<ViewType("CustomerListView")>
	Partial Public Class CustomersListView
		Inherits DevExpress.XtraBars.Ribbon.RibbonForm

		Public Sub New()
			InitializeComponent()
			If Not mvvmContext1.IsDesignMode Then
				InitializeBindings()
			End If
		End Sub
		Private Sub InitializeBindings()
			mvvmContext1.RegisterService(New InstantFeedbackService(CustomersGridView))
			'
			Dim fluentApi = mvvmContext1.OfType(Of CustomerListViewModel)()
			fluentApi.SetBinding(CustomersGridControl, Function(bs) bs.DataSource, Function(x) x.Customers)
			fluentApi.WithEvent(Of RowClickEventArgs)(CustomersGridView, NameOf(CustomersGridView.RowClick)).EventToCommand(Sub(x) x.Edit(), Function(args As RowClickEventArgs) args.Clicks = 2 AndAlso args.Button = MouseButtons.Left)
			fluentApi.BindCommand(btnNew, Sub(x) x.Add())
			fluentApi.BindCommand(btnDelete, Sub(x) x.Delete())
		End Sub
	End Class
End Namespace
